src/lit-html.ts

```diff
@@ -268,23 +268,34 @@ export class NodePart extends Part {
   setValue(value: any): void {
     value = this._getValue(value);
 
-    if (value instanceof Node) {
-      this._previousValue = this._setNode(value);
+    if (value === undefined ||
```

Comment:
Recommend documenting this branch. The rest are pretty clear but this one is doing a lot of subtle stuff.